File: solutions/go/sum-of-multiples/1/sum_of_multiples.go
package summultiples
func sumMultiples(m map[int]struct{}) int {
	var ksum int
	for k := range m {
		ksum += k
	}
	return ksum
}
func SumMultiples(limit int, divisors ...int) int {
	multiples := map[int]struct{}{}
	for _, d := range divisors {
		if d == 0 {
			continue
		}
		for i := d; i < limit; i += d {
			multiples[i] = struct{}{}
		}
	}
	return sumMultiples(multiples)
}

File: solutions/fortran/sum-of-multiples/1/sum_of_multiples.f90
! For my Shree DR.MDD  
module sum_of_multiples
   implicit none

contains

   function sum_multiples(factors, limit) result(outcome)
      integer, intent(in) :: factors(:), limit
      integer :: outcome
      integer :: pos, avail, step, val
      integer, dimension(:), allocatable :: store

      outcome = 0
      avail = 1
      allocate(store(limit))

      do pos = lbound(factors, dim=1), ubound(factors, dim=1)
         step = factors(pos)
         if (step == 0) cycle
         val = step
         do while (val < limit)
            if (.not. any(store(1:(avail - 1)) == val)) then
               store(avail) = val
               avail = avail + 1
               outcome = outcome + val
            end if
            val = val + step
         end do
      end do

   end function sum_multiples

end module sum_of_multiples
